pkg/kubelet/server/auth_test.go

```diff
@@ -82,60 +88,78 @@ const (
-		"/exec/{podNamespace}/{podID}/{uid}/{containerName}": "proxy",
-		"/healthz":                            "proxy",
-		"/healthz/log":                        "proxy",
-		"/healthz/ping":                       "proxy",
-		"/healthz/syncloop":                   "proxy",
-		"/logs/":                              "log",
-		"/logs/{logpath:*}":                   "log",
-		"/metrics":                            "metrics",
-		"/metrics/cadvisor":                   "metrics",
-		"/metrics/probes":                     "metrics",
-		"/metrics/resource":                   "metrics",
-		"/pods/":                              "proxy",
-		"/portForward/{podNamespace}/{podID}": "proxy",
-		"/portForward/{podNamespace}/{podID}/{uid}":         "proxy",
-		"/run/{podNamespace}/{podID}/{containerName}":       "proxy",
-		"/run/{podNamespace}/{podID}/{uid}/{containerName}": "proxy",
-		"/runningpods/":  "proxy",
-		"/stats/":        "stats",
-		"/stats/summary": "stats",
+	testPaths := map[string][]string{
+		"/attach/{podNamespace}/{podID}/{containerName}":       {"proxy"},
+		"/attach/{podNamespace}/{podID}/{uid}/{containerName}": {"proxy"},
+		"/checkpoint/{podNamespace}/{podID}/{containerName}":   {"checkpoint"},
+		"/configz": {"proxy"},
+		"/containerLogs/{podNamespace}/{podID}/{containerName}": {"proxy"},
+		"/debug/flags/v":                                     {"proxy"},
+		"/debug/pprof/{subpath:*}":                           {"proxy"},
+		"/exec/{podNamespace}/{podID}/{containerName}":       {"proxy"},
+		"/exec/{podNamespace}/{podID}/{uid}/{containerName}": {"proxy"},
+		"/healthz":                            {"proxy"},
+		"/healthz/log":                        {"proxy"},
+		"/healthz/ping":                       {"proxy"},
+		"/healthz/syncloop":                   {"proxy"},
+		"/logs/":                              {"log"},
+		"/logs/{logpath:*}":                   {"log"},
+		"/metrics":                            {"metrics"},
+		"/metrics/cadvisor":                   {"metrics"},
+		"/metrics/probes":                     {"metrics"},
+		"/metrics/resource":                   {"metrics"},
+		"/pods/":                              {"proxy"},
+		"/portForward/{podNamespace}/{podID}": {"proxy"},
+		"/portForward/{podNamespace}/{podID}/{uid}":         {"proxy"},
+		"/run/{podNamespace}/{podID}/{containerName}":       {"proxy"},
+		"/run/{podNamespace}/{podID}/{uid}/{containerName}": {"proxy"},
+		"/runningpods/":  {"proxy"},
+		"/stats/":        {"stats"},
+		"/stats/summary": {"stats"},
+	}
+
+	if fineGrained {
+		testPaths["/healthz"] = []string{"healthz", "proxy"}
+		testPaths["/healthz/log"] = []string{"healthz", "proxy"}
+		testPaths["/healthz/ping"] = []string{"healthz", "proxy"}
+		testPaths["/healthz/syncloop"] = []string{"healthz", "proxy"}
+		testPaths["/pods/"] = []string{"pods", "proxy"}
+		testPaths["/runningpods/"] = []string{"pods", "proxy"}
+		testPaths["/configz"] = []string{"configz", "proxy"}
```

Comment:
rather than overwriting the values from above, can we append the fine-grained subresource here, e.g. `testPaths["/healthz"] = append(testPaths["/healthz"], "healthz")`

Reply:
I thought of that too but it just reads weird, because we want to append the fine grained stuff at the beginning

```go
testPaths["/healthz"] = append([]string{"healthz"}, testPaths["/healthz"]...)
```

However I don't feel strongly about it.